src/utils/supAuth.js

```diff
@@ -0,0 +1,232 @@
+import * as files from './files';
+import ONA from '../connectors/ona-api/ona';
+
+// Default Authentications
+export const defaultOauthC = () => !!localStorage.getItem('access_token');
+
+export const defaultSupAuthC = (User, AuthConfig) => {
+  let user;
+  let authConfig;
+  // Define user
+  if (!User && !localStorage.getItem('user')) {
+    return false;
+  } else if (!User) {
+    user = JSON.parse(localStorage.getItem('user'));
+  } else {
+    user = { ...User };
+  }
+
+  // Define authConfig
+  if (!AuthConfig && !localStorage.getItem('authConfig')) {
+    return false;
+  } else if (!AuthConfig) {
+    authConfig = JSON.parse(localStorage.getItem('authConfig'));
+  } else {
+    authConfig = { ...AuthConfig };
+  }
+
+  if (defaultOauthC()) {
+    if (!authConfig || !authConfig.SITE || (authConfig.SITE && authConfig.SITE === 'public')) {
+      return true;
+    }
+    if (Array.isArray(authConfig.SITE)) {
```

Comment:
@cKellyDesign at what point does `authConfig.SITE` become an array of usernames? aren't we supposed to check the array of users with access to a particular view e.g `authConfig.VIEWS['/']`?